components/eventing-controller/api/v1alpha2/testing_helpers.go

```diff
@@ -0,0 +1,128 @@
+package v1alpha2
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	ProtocolSettingsContentModeBinary = "BINARY"
+	DefaultName                       = "test"
+	DefaultNamespace                  = "test-namespace"
+	DefaultSink                       = "https://svc2.test.local"
+	DefaultID                         = "id"
+	DefaultMaxInFlight                = 10
+	DefaultStatusReady                = true
+	DefaultConditions                 = []Condition{
+		{
+			Type:   ConditionSubscriptionActive,
+			Status: "true",
+		},
+		{
+			Type:   ConditionSubscribed,
+			Status: "false",
+		}}
+)
+
+// +kubebuilder:object:generate=false
+type SubscriptionOpt func(subscription *Subscription)
+
+func NewDefaultSubscription(opts ...SubscriptionOpt) *Subscription {
+	newSub := &Subscription{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Subscription",
+			APIVersion: "eventing.kyma-project.io/v1alpha2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DefaultName,
+			Namespace: DefaultNamespace,
+		},
+		Spec: SubscriptionSpec{
+			TypeMatching: EXACT,
+			Sink:         DefaultSink,
+			ID:           DefaultID,
+			Config: map[string]string{
+				MaxInFlightMessages: fmt.Sprint(DefaultMaxInFlight),
+			},
+		},
+		Status: SubscriptionStatus{
+			Ready:      DefaultStatusReady,
+			Conditions: DefaultConditions,
+		},
+	}
+	for _, o := range opts {
+		o(newSub)
+	}
+
+	return newSub
+}
+
+func WithSource(source string) SubscriptionOpt {
+	return func(sub *Subscription) {
+		sub.Spec.Source = source
+	}
+}
+
+func WithTypes(types []string) SubscriptionOpt {
+	return func(sub *Subscription) {
+		sub.Spec.Types = types
+	}
+}
+func WithStatusJetStreamTypes(types []JetStreamTypes) SubscriptionOpt {
+	return func(sub *Subscription) {
+		sub.Status.Backend.Types = types
+	}
+}
+
+func WithStatusTypes(statusTypes []EventType) SubscriptionOpt {
+	return func(sub *Subscription) {
+		if statusTypes == nil {
+			sub.Status.InitializeEventTypes()
+			return
+		}
+		sub.Status.Types = statusTypes
+	}
+}
+
+func WithWebhookAuthForBEB() SubscriptionOpt {
```

Comment:
The idea was to have non-shared testing methods to avoid the recursive imports in the both of the api packages. and when the v1alpha1 will be deprecated, we will get rid of it anyhow, so I think we could leave it as is.